rundetection/rules/loq_rules.py

```diff
@@ -0,0 +1,174 @@
+    return sample_title in sans_file.title and sans_file.type == "TRANS"
+
+
+def _is_sample_direct_file(sans_file: SansFileData) -> bool:
+    return ("direct" in sans_file.title.lower() or "empty" in sans_file.title.lower()) and sans_file.type == "TRANS"
+
+
+def _is_can_scatter_file(sans_file: SansFileData, can_title: str) -> bool:
+    return can_title == sans_file.title.split("_")[0] and sans_file.type == "SANS/TRANS"
+
+
+def _is_can_transmission_file(sans_file: SansFileData, can_title: str) -> bool:
+    return can_title in sans_file.title and sans_file.type == "TRANS"
+
+
+def _find_trans_file(sans_files: list[SansFileData], sample_title: str) -> SansFileData | None:
+    for sans_file in sans_files:
+        if _is_sample_transmission_file(sans_file=sans_file, sample_title=sample_title):
+            return sans_file
+    return None
+
+
+def _find_direct_file(sans_files: list[SansFileData]) -> SansFileData | None:
+    reversed_files = reversed(sans_files)
+    for sans_file in reversed_files:
+        if _is_sample_direct_file(sans_file=sans_file):
+            return sans_file
+    return None
+
+
+def _find_can_scatter_file(sans_files: list[SansFileData], can_title: str) -> SansFileData | None:
+    for sans_file in sans_files:
+        if _is_can_scatter_file(sans_file=sans_file, can_title=can_title):
+            return sans_file
+    return None
+
+
+def _find_can_trans_file(sans_files: list[SansFileData], can_title: str) -> SansFileData | None:
+    for sans_file in sans_files:
+        if _is_can_transmission_file(sans_file=sans_file, can_title=can_title):
+            return sans_file
+    return None
+
+
+def find_path_for_run_number(cycle_path: str, run_number: int) -> Path | None:
+    # 10 is just a magic number, but we needed an unrealistic value for the maximum
+    for padding in range(1, 11):
+        potential_path = Path(f"{cycle_path}/LOQ{str(run_number).zfill(padding)}.nxs")
+        if potential_path.exists():
+            return potential_path
+    return None
+
+
+def grab_cycle_instrument_index(cycle: str) -> str:
+    _, cycle_year, cycle_num = cycle.split("_")
+    url = f"http://data.isis.rl.ac.uk/journals/ndxloq/journal_{cycle_year}_{cycle_num}.xml"
+    return requests.get(url, timeout=5).text
+
+
+def create_list_of_files(job_request: JobRequest) -> list[SansFileData]:
+    cycle = _extract_cycle_from_file_path(file_path=job_request.filepath)
+    xml = grab_cycle_instrument_index(cycle=cycle)
+    cycle_run_info = xmltodict.parse(xml)
+    list_of_files = []
+    for run_info in cycle_run_info["NXroot"]["NXentry"]:
+        title_contents = run_info["title"]["#text"].split("_")
+        run_number = run_info["run_number"]["#text"]
+        if len(title_contents) == 2:  # noqa: PLR2004
+            _, file_type = title_contents
+        elif len(title_contents) == 3:  # noqa: PLR2004
+            _, __, file_type = title_contents
```

Comment:
As a suggestion, instead of using multiple conditions and throwaway variables, can check the length in 1 line and then get the last element of the list using `[-1]`
```suggestion
    if len(title_contents) in {2, 3}: 
            file_type = title_contents[-1]
```